src/wallet/coinselection.cpp

```diff
@@ -0,0 +1,370 @@
+// how many satoshi's the coin selection is willing to be off by per iteration through the loop.  Makes it progressively
+// easier to find a solution
+// (now its a simple exponential) const long int LOOP_COST = 500;
+const int P2PKH_INPUT_SIZE = 72; // <Sig> <PubKey> OP_DUP OP_HASH160 <PubkeyHash> OP_EQUALVERIFY OP_CHECKSIG
+
+extern CTweak<unsigned int> maxCoinSelSearchTime;
+extern CTweak<unsigned int> preferredNumUTXO;
+
+#if 0 // for reference. Defined in wallet.h
+typedef std::multimap<CAmount, COutput> SpendableTxos;
+typedef std::set<std::pair<const CWalletTx*,unsigned int> > CoinSet;
+typedef std::set<SpendableTxos::const_iterator> TxoItVec;
+typedef std::pair<CAmount, TxoItVec > TxoGroup;  // A set of coins and how much they sum to.
+
+struct TxoIterLess // : binary_function <T,T,bool> 
+  {
+    typedef SpendableTxos::const_iterator T;
+    bool operator() (const T& x, const T& y) const {return x->first < y->first;}
+  };
+
+typedef std::set<SpendableTxos::iterator,TxoIterLess> TxoItVec;
+typedef std::pair<CAmount, TxoItVec > TxoGroup;  // A set of coins and how much they sum to.
+#endif
+
+// This holds a sorted map of coin amounts and the set of Txos required to get to that amount
+typedef std::multimap<CAmount, TxoGroup> TxoGroupMap;
+
+// How close do we want to get to targetValue (at first).  The longer it takes the more we add to this value.
+CAmount reasonableExcess(CAmount targetValue) { return targetValue / 1024; }
+// Create a group of Txos from a Txo index and another group.
+TxoGroup makeGroup(SpendableTxos::iterator i, const TxoGroup *prev = nullptr)
+{
+    TxoGroup ret;
+    if (prev != nullptr)
+        ret = *prev;
+    else
+        ret.first = 0;
+    // Don't add an element that already exists  -- allow this, bad solutions eliminated later
+    // assert(ret.second.find(i) == ret.second.end());
+    ret.first += i->first; // update the amount
+    COutput outp = i->second;
+    // verify that the output value is consistent -- TODO: Check pcoin and make sure it doesn't get deleted -- located
+    // in mapWallet
+    assert(outp.tx->vout[outp.i].nValue == i->first);
+    ret.second.insert(i); // add this txo to the set
+    return ret;
+}
+
+// Take a group that is not a solution and find a bunch of solutions by appending additional TXOs onto it.
+// This is a recursive function that is limited by MAX_TXOs and MAX_ELECTIVE_TXOS
+bool extend(const CAmount targetValue,
+    /*const*/ SpendableTxos &available,
+    TxoGroup grp,
+    TxoGroupMap &solutions,
+    int depth)
+{
+    if (depth >= MAX_TXOS)
+        return false; // unambiguously break the recursion
+    SpendableTxos::iterator aEnd = available.end(); // For speed, get these once
+    SpendableTxos::iterator aBeg = available.begin();
+    bool ret = false;
+    SpendableTxos::iterator small;
+    small = available.lower_bound(targetValue - grp.first); // Find a value at or above what we need
+    if (1)
+    {
```

Comment:
do you still need the if(1)?

Reply:
no sorry, that's the old style I used for opening a new scope.